Debug `dotnet restore` problems

dougbu opened the pull request:
- remove `--ignore-failed-sources`
- use `dotnet trace` to get more data

dougbu committed:
Remove `--ignore-failed-sources`

dougbu committed:
Use `dotnet trace` to get more data

kartheekp-ms commented on eng/helix/content/runtests.cmd:
```diff
@@ -23,12 +23,22 @@ echo.
+)
+
+echo "Restore: dotnet restore RunTests\RunTests.csproj"
+dotnet trace collect --providers "System.Net.Http,System.Net.Sockets,System.Net.Security,System.Net.NameResolution" -- dotnet restore RunTests\RunTests.csproj --verbosity detailed
```
I don't see `-- dotnet restore RunTests/RunTests.csproj --verbosity detailed` in the trace command. Is this expected?

dougbu committed:
!fixup! Install `dotnet-trace` in work item root
- global dotnet tool location is not in `$env:PATH` by default

dougbu committed:
!fixup! Avoid restore errors when installing tool
- `dotnet-trace` installation hit same `dotnet restore` issue in last iteration

dougbu commented:
/azp run

dougbu committed:
!fixup! Remove Windows.10.Arm64v8.Open queue
- `dotnet-trace` is not supported on Windows ARM64
